src/uchen/tboard/tboard_file.h

```diff
@@ -2,24 +2,46 @@
 #define UCHEN_TBOARD_TBOARD_FILE_H_
 
 #include <fstream>
+#include <string>
 #include <string_view>
 
 #include "absl/status/statusor.h"
 
+#include "proto/summary.pb.h"
+
 namespace uchen::tboard {
 
+// MaskDelta is a fixed arbitrary number that is used to mask the CRC32
+// checksum. expected from TensorBoard.
+constexpr uint32_t kMaskDelta = 0xa282ead8ul;
+
+constexpr std::string_view kLogKey = "tfevents";
+
 class TBoardFile {
  public:
   static absl::StatusOr<TBoardFile> Open(std::string_view path);
 
   TBoardFile(TBoardFile&&) = default;
-  ~TBoardFile() = default;
+  ~TBoardFile();
+
+  void RecordLoss(double loss, uint32_t step);
+  void RecordScalar(const std::string& tag, double value, uint32_t step);
 
-  void RecordLoss(float loss);
+  void RecordImage(const std::string& tag, const std::string& image,
+                   uint32_t step, uint32_t width, uint32_t height,
+                   uint32_t channel, const std::string& image_name,
+                   const std::string& image_description);
 
  private:
   TBoardFile(std::ofstream file) : file_(std::move(file)) {}
 
+  void AddScalar(const std::string& tag, double value, uint32_t step);
```

Comment:
```suggestion
  void AddScalar(std::string_view tag, double value, uint32_t step);
```
Can also be made public, replacing `RecordLoss` and `RecordScalar`

Reply:
Done

Reply:
The usage of std::string_view cannot be done in the functions deals directly with `TensorFlow` API as it cannot be passed as arguments for the functions. and we will end up explicitly casting to std::string preventing benefiting of string_view usage.

Reply:
`set_tag` takes rvalue reference and copies anyway. Constructing std::string when calling it is fine. Having std::string_view in API has a lot of benefits as it allows callers use whatever string they like, not just `std::string`.